src/app/shared/services/task/task.service.ts

```diff
@@ -1,26 +1,117 @@
 import { Injectable } from "@angular/core";
+import { IAppConfig } from "packages/data-models";
 import { TemplateFieldService } from "../../components/template/services/template-field.service";
+import { AppConfigService } from "../app-config/app-config.service";
+import { AppDataService } from "../data/app-data.service";
+import clone from "clone";
+import { arrayToHashmap } from "../../utils";
 
 @Injectable({
   providedIn: "root",
 })
 export class TaskService {
-  constructor(private templateFieldService: TemplateFieldService) {}
+  appFields: IAppConfig["APP_FIELDS"];
+  taskGroups: any[];
+  taskGroupsHashmap: Record<string, any>;
+  constructor(
+    private templateFieldService: TemplateFieldService,
+    private appConfigService: AppConfigService,
+    private appDataService: AppDataService
+  ) {
+    this.subscribeToAppConfigChanges();
+  }
+
+  async init() {
+    await this.getListOfTaskGroups();
+    this.evaluateHighlightedTaskGroup();
+  }
+
+  /** Get the list of highlight-able task groups, from the relevant data_list */
+  private async getListOfTaskGroups() {
+    // TODO: This should be set from the template/skin level
+    const taskGroupsListName = "workshop_tasks";
+    const taskGroupsDataList = await this.appDataService.getSheet("data_list", taskGroupsListName);
+    this.taskGroups = taskGroupsDataList.rows;
+    this.taskGroupsHashmap = arrayToHashmap(this.taskGroups, "id");
+  }
+
+  /** The _highlighted_task_group should always be the ID of the highest
+   * priority task_group that is not completed and not skipped
+   * NB "highest priority" is defined as having the lowest numerical value for the "number" column
+   */
+  public async evaluateHighlightedTaskGroup() {
+    const taskGroupsNotCompletedAndNotSkipped = clone(this.taskGroups).filter((taskGroup) => {
+      return (
+        !this.templateFieldService.getField(taskGroup.completed_field) &&
+        !this.templateFieldService.getField(taskGroup.skipped_field)
+      );
+    });
+    // If all task groups are completed or skipped (e.g. when user completes final task group),
+    // then set highlighted task group to undefined
+    if (taskGroupsNotCompletedAndNotSkipped.length === 0) {
+      this.templateFieldService.setField(this.appFields.HIGHLIGHTED_DATA_ID, "undefined");
```

Comment:
nit(non-blocking)
Setting string undefined feels weird, can it just be set as a blank string instead?